builder/ibmcloud/vpc/step_create_instance.go

```diff
@@ -24,104 +24,171 @@ func (step *stepCreateInstance) Run(_ context.Context, state multistep.StateBag)
 
 	vsiBaseImageName := config.VSIBaseImageName
 	vsiBaseImageID := config.VSIBaseImageID
+	vsiCatalogOfferingCrn := config.CatalogOfferingCRN
+	vsiCatalogOfferingVersionCrn := config.CatalogOfferingVersionCRN
 
-	keyIDentityModel := &vpcv1.KeyIdentityByID{
+	keyIdentityModel := &vpcv1.KeyIdentityByID{
 		ID: &[]string{state.Get("vpc_ssh_key_id").(string)}[0],
 	}
 	instanceProfileIdentityModel := &vpcv1.InstanceProfileIdentityByName{
 		Name: &[]string{config.VSIProfile}[0],
 	}
-	vpcIDentityModel := &vpcv1.VPCIdentityByID{
+	vpcIdentityModel := &vpcv1.VPCIdentityByID{
 		ID: &[]string{state.Get("vpc_id").(string)}[0],
 	}
-	subnetIDentityModel := &vpcv1.SubnetIdentityByID{
+	subnetIdentityModel := &vpcv1.SubnetIdentityByID{
 		ID: &[]string{config.SubnetID}[0],
 	}
 	networkInterfacePrototypeModel := &vpcv1.NetworkInterfacePrototype{
 		Name:   &[]string{"my-instance-modified"}[0],
-		Subnet: subnetIDentityModel,
+		Subnet: subnetIdentityModel,
 	}
 	zoneIdentityModel := &vpcv1.ZoneIdentityByName{
 		Name: &[]string{state.Get("zone").(string)}[0],
 	}
 
 	ui.Say("Creating Instance...")
 
-	// Get Image ID
-	if vsiBaseImageName != "" {
-		ui.Say("Fetching ImageID...")
+	// For catalog images
+	if vsiCatalogOfferingCrn != "" || vsiCatalogOfferingVersionCrn != "" {
```

Comment:
suggestive change to use CRN Vs Crn as it is a widely used acronym
```suggestion
	if vsiCatalogOfferingCRN != "" || vsiCatalogOfferingVersionCrn != "" {
```